app/forms/fields.py

```diff
@@ -124,15 +124,15 @@ class CustomIntegerField(IntegerField):
     further validation steps by using a separate IntegerCheck validator
     """
     def __init__(self, **kwargs):
-
+        self.data = None
         super(CustomIntegerField, self).__init__(**kwargs)
 
     def process_formdata(self, valuelist):
         if valuelist:
             try:
                 self.data = int(valuelist[0])
             except ValueError:
-                self.data = None
+                pass
```

Comment:
Is it possible that self.data could contain its previous value if it isn't reset?